conda_forge_tick/migrators/version.py

```diff
@@ -636,55 +636,60 @@ def pr_body(self, feedstock_ctx: FeedstockContext) -> str:
-                    f"Analysis of the source code shows a discrepancy between"
-                    f" the library's imports and the package's stated requirements"
-                    f" in the meta.yaml."
+                    "Please note that this analysis is **highly experimental**. "
+                    "The aim here is to make maintenance easier by inspecting the package's dependencies. "
+                    "Importantly this analysis does not support optional dependencies, "
+                    "please double check those before making changes. "
+                    "If you do not want hinting of this kind ever please add "
+                    "`bot: inspection: false` to your `conda-forge.yml`. "
+                    "If you encounter issues with this feature please ping the bot team `conda-forge/bot`.\n\n"
                 )
-                if df_cf:
+                if dep_comparison:
+                    df_cf = ""
+                    for k in dep_comparison.get("df_minus_cf", set()):
+                        df_cf += f"- {k}" + "\n"
+                    cf_df = ""
+                    for k in dep_comparison.get("cf_minus_df", set()):
+                        cf_df += f"- {k}" + "\n"
                     hint += (
-                        f"\n\n### Packages found by inspection but not in the meta.yaml:\n"
-                        f"{df_cf}"
+                        f"Analysis of the source code shows a discrepancy between"
+                        f" the library's imports and the package's stated requirements"
+                        f" in the meta.yaml."
                     )
-                if cf_df:
+                    if df_cf:
+                        hint += (
+                            f"\n\n### Packages found by inspection but not in the meta.yaml:\n"
+                            f"{df_cf}"
+                        )
+                    if cf_df:
+                        hint += (
+                            f"\n\n### Packages found in the meta.yaml but not found by inspection:\n"
+                            f"{cf_df}"
+                        )
+                else:
                     hint += (
-                        f"\n\n### Packages found in the meta.yaml but not found by inspection:\n"
-                        f"{cf_df}"
+                        "Analysis of the source code shows **no** discrepancy between"
+                        " the library's imports and the package's stated requirements in the meta.yaml."
                     )
-            else:
-                hint += (
-                    "Analysis of the source code shows **no** discrepancy between"
-                    " the library's imports and the package's stated requirements in the meta.yaml."
-                )
+                body += hint
+        except Exception:
+            hint = f"\n\nDependency Analysis\n--------------------\n\n"
+            hint += "We couldn't run dependency analysis due to an error.\n"
```

Comment:
```suggestion
            hint += (
                "We couldn't run dependency analysis due to an error. Make sure the "
                "recipe and the source code download URL are correct.\n"
            )
```

Reply:
That's a bad description for the original issue that prompted this.

Reply:
ack I missed the patch :/